wcfsetup/install/files/lib/system/package/plugin/AbstractFileDeletePackageInstallationPlugin.class.php

```diff
@@ -130,6 +129,47 @@ protected function handleDelete(array $items)
         WCF::getDB()->commitTransaction();
     }
 
+    private static function isFilesystemCaseSensitive(): bool
+    {
+        static $isFilesystemCaseSensitive = null;
+
+        if ($isFilesystemCaseSensitive === null) {
+            $testFilePath = __FILE__;
+
+            // Toggles the case of the filePath: Example -> eXAMPLE
+            // https://stackoverflow.com/a/6612519
+            $invertedCase = \mb_strtolower($testFilePath) ^ \mb_strtoupper($testFilePath) ^ $testFilePath;
+
+            return !\file_exists($invertedCase);
```

Comment:
Shouldn't the result of this expression be cached in `$isFilesystemCaseSensitive` prior to returning? :)